src/appleseed/renderer/kernel/rendering/renderercomponents.h

```diff
@@ -86,28 +86,30 @@ class RendererComponents
-    ITileCallbackFactory*                               m_tile_callback_factory;
-    const Scene&                                        m_scene;
-    const Frame&                                        m_frame;
-    const TraceContext&                                 m_trace_context;
-    std::unique_ptr<ForwardLightSampler>                m_forward_light_sampler;
-    std::unique_ptr<BackwardLightSampler>               m_backward_light_sampler;
-    ShadingEngine                                       m_shading_engine;
-    TextureStore&                                       m_texture_store;
-    OIIOTextureSystem&                                  m_texture_system;
-    OSLShadingSystem&                                   m_shading_system;
-
-    std::unique_ptr<ILightingEngineFactory>             m_lighting_engine_factory;
-    std::unique_ptr<ISampleRendererFactory>             m_sample_renderer_factory;
-    std::unique_ptr<ISampleGeneratorFactory>            m_sample_generator_factory;
-    std::unique_ptr<IPixelRendererFactory>              m_pixel_renderer_factory;
-    std::unique_ptr<IShadingResultFrameBufferFactory>   m_shading_result_framebuffer_factory;
-    std::unique_ptr<ITileRendererFactory>               m_tile_renderer_factory;
-    std::unique_ptr<IPassCallback>                      m_pass_callback;
-    foundation::auto_release_ptr<IFrameRenderer>        m_frame_renderer;
+    const Project&                                                  m_project;
+    const ParamArray&                                               m_params;
+    ITileCallbackFactory*                                           m_tile_callback_factory;
+    const Scene&                                                    m_scene;
+    const Frame&                                                    m_frame;
+    const TraceContext&                                             m_trace_context;
+    std::unique_ptr<ForwardLightSampler>                            m_forward_light_sampler;
+    std::unique_ptr<BackwardLightSampler>                           m_backward_light_sampler;
+    ShadingEngine                                                   m_shading_engine;
+    TextureStore&                                                   m_texture_store;
+    OIIOTextureSystem&                                              m_texture_system;
+    OSLShadingSystem&                                               m_shading_system;
+
+    std::unique_ptr<ILightingEngineFactory>                         m_lighting_engine_factory;
+    std::unique_ptr<ISampleRendererFactory>                         m_sample_renderer_factory;
+    std::unique_ptr<ISampleGeneratorFactory>                        m_sample_generator_factory;
+    std::unique_ptr<IPixelRendererFactory>                          m_pixel_renderer_factory;
+    foundation::auto_release_ptr<IShadingResultFrameBufferFactory>  m_shading_result_framebuffer_factory;
```

Comment:
Why this change?

Reply:
Because the ptr is not unique anymore.

Reply:
The master renderer use it.

Reply:
That's not a valid argument :) `foundation::auto_release_ptr<>` is like the now-deprecated `std::auto_ptr<>`, which is the ancestor of `std::unique_ptr<>`. All three are "unique pointers".

The only relevant difference between `foundation::auto_release_ptr<>` and `std::unique_ptr<>` is that the former calls a `release()` method to destroy the object while the later calls the destructor.

In this context, either points would work. I don't see any reason to change the type in this PR.

Reply:
I didn't knew that :o The names aren't really helping.